### PR TITLE
Install config files in correct location

### PR DESCRIPTION
**Pull request by hansfn**
According to https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure the config files should be *not* placed under /usr/share (which is what DATA_INSTALL_DIR gives you). Putting it one location under /usr/share makes it impossible to install for multiple libs (or archs).

The provided patch is only tested under Linux and might need extension to work under Windows. See link above for Windows only search path. The Unix search paths should work for Windows too.

PS! This patch uses LIB_INSTALL_DIR/PROJECTNAME/cmake and not LIB_INSTALL_DIR/cmake/PROJECTNAME, which I think is more common on Linux, in the hope that it will work under Windows too.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/52/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/56/)

**Comment by hansfn** [comment]
Updated for 1.0.11.

This fix is still needed for multilib installs since the config files aren't identical for lib32 and lib64.

File: src/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -137,7 +137,7 @@ if(MSVC OR MSVC_IDE AND EPROSIMA_INSTALLER)
     endif()
 
     install(FILES ${PROJECT_SOURCE_DIR}/cmake/packaging/windows/${PROJECT_NAME}-config.cmake
-        DESTINATION ${DATA_INSTALL_DIR}/${PROJECT_NAME}/cmake
+        DESTINATION ${LIB_INSTALL_DIR}/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT}
```
Comment (richiware) [inline src/cpp/CMakeLists.txt]:
```suggestion
        DESTINATION ${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}${MSVCARCH_DIR_EXTENSION_EXT}
```

File: src/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -154,7 +154,7 @@ elseif(NOT EPROSIMA_INSTALLER)
     if(INSTALLER_PLATFORM)
         set(INSTALL_DESTINATION_PATH ${DATA_INSTALL_DIR}/${PROJECT_NAME}-${INSTALLER_PLATFORM}/cmake)
     else()
-        set(INSTALL_DESTINATION_PATH ${DATA_INSTALL_DIR}/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT})
+        set(INSTALL_DESTINATION_PATH ${LIB_INSTALL_DIR}/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT})
```
Comment (richiware) [inline src/cpp/CMakeLists.txt]:
```suggestion
        set(INSTALL_DESTINATION_PATH ${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}${MSVCARCH_DIR_EXTENSION_EXT})
```

File: src/cpp/CMakeLists.txt (hunk 1 above)
Comment (hansfn) [reply src/cpp/CMakeLists.txt]: I have no objections to this change.

**Comment by hansfn** [comment]
I'm a Linux user and prefer lib/cmake/${projectname} myself, but I tried to be kind to Windows users - ref PS at the end of the PR. (Not sure if I resolved your suggestions correctly.)

File: src/cpp/CMakeLists.txt (hunk 2 above)
Comment (hansfn) [reply src/cpp/CMakeLists.txt]: I have no objections to this change either ;-)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/9/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/10/)

**Comment by richiware** [comment]
You remind me of something which has to be also changed. `MSVCARCH_DIR_EXTENSION_EXT` is used in a special case, when we generate a windows installer with several platforms. Those CMake config files are called by a parent CMake config file. The template is located in `cmake/packaging/windows/fastcdr-config.cmake` and now it should be:

```cmake
if(MSVC_VERSION EQUAL 1900)
    if(CMAKE_CL_64)
        include("${CMAKE_CURRENT_LIST_DIR}/../lib/cmake/fastcdr-x64Win64VS2015/fastcdr-config.cmake")
    else()
        include("${CMAKE_CURRENT_LIST_DIR}/../lib/cmake/fastcdr-i86Win32VS2015/fastcdr-config.cmake")
    endif()
elseif(MSVC_VERSION GREATER 1900)
    if(CMAKE_CL_64)
        include("${CMAKE_CURRENT_LIST_DIR}/../lib/cmake/fastcdr-x64Win64VS2017/fastcdr-config.cmake")
    else()
        include("${CMAKE_CURRENT_LIST_DIR}/../lib/cmake/fastcdr-i86Win32VS2017/fastcdr-config.cmake")
    endif()
else()
    message(FATAL_ERROR "Not supported version of Visual Studio")
endif()
```
Can you make the changes, please?

**Comment by hansfn** [comment]
Done. Sorry about the delay.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/10/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/11/)

**Comment by richiware** [comment]
Thanks for the contribution.

**Pull request merged**